Lib/test/test_smtplib.py

```diff
@@ -1072,6 +1072,18 @@ def test_send_unicode_without_SMTPUTF8(self):
         self.assertRaises(UnicodeEncodeError, smtp.sendmail, 'Alice', 'Böb', '')
         self.assertRaises(UnicodeEncodeError, smtp.mail, 'Älice')
 
+    def test_remove_name_field_from_enveloppe_SMTPUTF8(self):
```

Comment:
I would name this "test_name_field_not_included_in_envelope_addresses", and check the addresses seen by the server.  That means changing SimSMTPServer's process_message method to record the from and to addresses so you can check them in the test method.